### PR TITLE
several improvements

### PR DESCRIPTION
**Pull request by froque**
- Upgrade parent pom to 3.51 and java level to 8
- removed unused class PluginsPerJob
- separate section for non-jobs plugins (related [JENKINS-58583](https://issues.jenkins-ci.org/browse/JENKINS-58583))
- Sort plugins and jobs by name (related [JENKINS-53269](https://issues.jenkins-ci.org/browse/JENKINS-53269))
- use full display name to show jobs with same name (related [JENKINS-53268](https://issues.jenkins-ci.org/browse/JENKINS-53268), [JENKINS-53267](https://issues.jenkins-ci.org/browse/JENKINS-53267))
- conditional build step (related [JENKINS-30673](https://issues.jenkins-ci.org/browse/JENKINS-30673))
- project parameters for plugins (related [JENKINS-48978](https://issues.jenkins-ci.org/browse/JENKINS-48978))
- promoted builds plugin (related [JENKINS-35159](https://issues.jenkins-ci.org/browse/JENKINS-35159))
- added Intellij files to .gitignore
- Added declarative pipeline support for step functions (related [JENKINS-53597](https://issues.jenkins-ci.org/browse/JENKINS-53597))
- Added Jenkinsfile to enable builds on https://ci.jenkins.io/
- Added Maven projects (related [JENKINS-41507](https://issues.jenkins-ci.org/browse/JENKINS-41507))

### CODE REVIEW

**Comment by StefanSpieker** [comment]
Wow, I like these additions. Hopefully it gets merged soon.

**Comment by froque** [comment]
@michallaskowski @chrissy25 can any of you review this pull request ?

File: src/main/java/org/jenkinsci/plugins/pluginusage/analyzer/BuildWrapperJobAnalyzer.java [hunk 1]
```diff
@@ -25,11 +26,12 @@ public BuildWrapperJobAnalyzer() {
...
 	}
-	
-	protected void doJobAnalyze(AbstractProject item, HashMap<PluginWrapper, JobsPerPlugin> mapJobsPerPlugin)
+
+	@Override
+	protected void doJobAnalyze(Job item, Map<PluginWrapper, JobsPerPlugin> mapJobsPerPlugin)
 	{
 		super.doJobAnalyze(null, mapJobsPerPlugin);
-		if (item instanceof BuildableItemWithBuildWrappers)
+		if (item != null && item instanceof BuildableItemWithBuildWrappers)
```
Comment (StefanSpieker) [inline src/main/java/org/jenkinsci/plugins/pluginusage/analyzer/BuildWrapperJobAnalyzer.java]: It is not needed to check for null. The instanceof will also work if item is null. 
null instanceof anything will be false.

**Comment by froque** [comment]
@StefanSpieker thanks for the input. changed it as commented

**Comment by michallaskowski** [comment]
> @michallaskowski @chrissy25 can any of you review this pull request ?

Sure. Thanks for all the work. I am catching up on changes. I will try to review and check it out tomorrow.

File: src/main/resources/org/jenkinsci/plugins/pluginusage/PluginUsageView/index.jelly [hunk 2]
```diff
@@ -57,6 +58,36 @@
...
+                        <th>
+                            ${%Version}
+                        </th>
+                    </tr>
+                </thead>
+                <tbody>
+                    <j:forEach var="j" items="${otherPlugins}">
+                        <tr>
+                            <td align="center" style="vertical-align:middle;">
+                                <a href="${j.plugin.url}">${j.longName}</a>
```
Comment (michallaskowski) [inline src/main/resources/org/jenkinsci/plugins/pluginusage/PluginUsageView/index.jelly]: The link doesn't work, it should be just `${j.url}`

File: src/main/java/org/jenkinsci/plugins/pluginusage/JobsPerPlugin.java [hunk 3]
```diff
@@ -2,29 +2,32 @@
 
 import hudson.PluginWrapper;
 import hudson.model.AbstractProject;
```
Comment (michallaskowski) [inline src/main/java/org/jenkinsci/plugins/pluginusage/JobsPerPlugin.java]: Seems to be an unused import

Comment (froque) [reply src/main/java/org/jenkinsci/plugins/pluginusage/JobsPerPlugin.java]: removed all unused imports

File: src/main/resources/org/jenkinsci/plugins/pluginusage/PluginUsageView/index.jelly (hunk 2 above)
Comment (froque) [reply src/main/resources/org/jenkinsci/plugins/pluginusage/PluginUsageView/index.jelly]: nice catch, thanks.  corrected.

**Comment by michallaskowski** [comment]
Thanks for all the hard work @froque  :)

**Pull request merged**